hoverxref/_static/js/hoverxref.js_t

```diff
@@ -96,9 +96,22 @@ $(document).ready(function() {
     // Remove ``title=`` attribute for intersphinx nodes that have hoverxref enabled.
     // It doesn't make sense the browser shows the default tooltip (browser's built-in)
     // and immediately after that our tooltip was shown.
-    $('.{{ hoverxref_css_class_prefix }}hoverxref.external').each(function () { $(this).removeAttr('title') });
 
-    $('.{{ hoverxref_css_class_prefix }}hoverxref.{{ hoverxref_css_class_prefix }}tooltip').tooltipster({
+    // Support lazy-loading here by switching between on-page-load (calling .tooltipster directly)
+    // or delaying it until after a mouseenter or click event.
+    // On large pages (the use case for lazy-loaded tooltipster) this moves dom manipulation to
+    // on-interaction, which is known to be less performant on small pages (as per tooltipster docs),
+    // but on massive docs pages fixes an otherwise severe page load stall when tooltipster
+    // manipulates the html for every single tooltip at once.
+    {% if not hoverxref_tooltip_lazy %}
+    $('.{{ hoverxref_css_class_prefix }}hoverxref.external').each(function () { $(this).removeAttr('title') });
+    $('.{{ hoverxref_css_class_prefix }}hoverxref.{{ hoverxref_css_class_prefix }}tooltip')
+    {% else %}
+    $('.{{ hoverxref_css_class_prefix }}hoverxref.{{ hoverxref_css_class_prefix }}tooltip').one('mouseenter click', function(event) {
+    $(this).removeAttr('title');
+    $(this)
+    {% endif %}
```

Comment:
Probably makes sense to invert the logic here and start with `if hoerxref_tooltip_lazy` first, which is the easier to read case.